### PR TITLE
[WIP] Introducing `gather_array`

### PR DESCRIPTION
**Pull request by ghislainb**
This is a first working version. Code simplification and comments are still needed.
Current limitation (that could be removed in the future) : local arrays should have the same size on all processes.

### CODE REVIEW

File: example/demo_gather.jl [hunk 1]
```diff
@@ -0,0 +1,60 @@
+"""
+Run this 2D-demo with 4 procs : mpirun -n 4 julia demo_2D.jl
+
+This demo synchronizes only one array. To synchronize multiple arrays, just use the same `update_halo!` method
+with a `Tuple` of `Array`.
+"""
+module demo
+include(string(@__DIR__, "/../src/MPICartesianGrid.jl"))
+using .MPICartesianGrid
+#using MPICartesianGrid
```
Comment (bmxam) [inline example/demo_gather.jl:10]: can you please uncomment this line and delete the two lines above?

File: src/grid.jl [hunk 2]
```diff
@@ -279,4 +279,71 @@ function _is_true_neighbor(neighbor::NTuple{D,Int}, stencil::NTuple{D,Int}, grid
     return all(d -> 1 <= neighbor[d] <= get_ndims(grid,d), 1:D) && any(d -> stencil[d] != 0, 1:D)
 end
 
-finalize_grid(::DistributedGrid; finalize_MPI = true) = finalize_MPI && MPI.Finalize()
\ No newline at end of file
+finalize_grid(::DistributedGrid; finalize_MPI = true) = finalize_MPI && MPI.Finalize()
+
+function gather_array(array::AbstractArray{T,D}, grid::DistributedGrid{D}, root::Integer = 0) where {T,D}
```
Comment (bmxam) [inline src/grid.jl:284]: Can you add the following documentation:
```
"""
    gather_array(array::AbstractArray{T,D}, grid::DistributedGrid{D}, root::Integer = 0) where {T,D}

Establish parallel communication to gather a distributed array on the `root` processor.
"""
```

**Review by bmxam — CHANGES_REQUESTED** [review]
is it too much work to add version for a ntuple of AbstractArray? (to exchange multiple "variables" in one call)?

Maybe just a dumb thing like `gather_array(t::NTuple) = map(gather_array, t)`?

File: src/grid.jl (hunk 2 above)
Comment (ghislainb) [reply src/grid.jl:290]: Done

**Review by bmxam — APPROVED** [review]
Thank you for this outstanding contribution to MPICartesianGrid.jl

**Pull request merged**